tests/Ticket/GH2783Test.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Tests\Ticket;
+
+use MongoDB\Laravel\Eloquent\Model;
+use MongoDB\Laravel\Relations\MorphTo;
+use Illuminate\Database\Eloquent\Relations\MorphOne;
+
+/**
+ * @see https://github.com/mongodb/laravel-mongodb/issues/2783
+ * @see https://jira.mongodb.org/browse/PHPORM-175
+ */
+class GH2783Test extends TestCase
+{
+    public function testMorphToInfersCustomOwnerKey()
+    {
+        GH2783Image::truncate();
+        GH2783Post::truncate();
+        GH2783User::truncate();
+
+        $post = GH2783Post::create(['text' => 'Lorem ipsum']);
+        $user = GH2783User::create(['username' => 'jsmith']);
+
+        $imageWithPost = GH2783Image::create(['uri' => 'http://example.com/post.png']);
+        $imageWithPost->imageable()->associate($post)->save();
+
+        $imageWithUser = GH2783Image::create(['uri' => 'http://example.com/user.png']);
+        $imageWithUser->imageable()->associate($user)->save();
+
+        $queriedImageWithPost = GH2783Image::with('imageable')->find($imageWithPost->getKey());
+        $this->assertInstanceOf(GH2783Post::class, $queriedImageWithPost->imageable);
+        $this->assertEquals($post->_id, $queriedImageWithPost->imageable->getKey());
+
+        $queriedImageWithUser = GH2783Image::with('imageable')->find($imageWithUser->getKey());
+        $this->assertInstanceOf(GH2783User::class, $queriedImageWithUser->imageable);
+        $this->assertEquals($user->username, $queriedImageWithUser->imageable->getKey());
+    }
+}
+
+class GH2783Image extends Model
```

Comment:
The CS error can be removed with this rule: https://github.com/mongodb/mongo-php-library/blob/fdfc1e74b39ab943329b986f0fe9f46eddb771d0/phpcs.xml.dist#L156-L159